Add patch version release criterion (#17546)
etcd-io/etcd: Documentation/contributor-guide/release.md

- Create new stable branch through `git push origin ${VERSION_MAJOR}.${VERSION_MINOR}` if this is a major stable release. This assumes `origin` corresponds to "https://github.com/etcd-io/etcd".
- Bump [hardcoded Version in the repository](https://github.com/etcd-io/etcd/blob/v3.4.15/version/version.go#L30) to the version `${VERSION}+git`.

> ## Patch release criterion

Reviewer: Nit: Simpler english suggestion

```suggestion
## Patch release criteria
```

Author: done & thx

> Consider releasing a patch version if any of the following conditions stand:

Reviewer: Suggest strengthening the language a bit.

```suggestion
The etcd project aims to release a new patch version if any of the following conditions are met:
```

Author: Done.

- Fixed one or more major CVEs (>=7.5).
- Fixed one or more critical bugs.
- Fixed three or more major bugs.

> - Fixed five or more minor bugs.

Reviewer: We were in a situation last year where we had backported adopting new golang minor version and it took us quite a while to cut a patch release making it available for users. For some users/orgs using software compiled with modern / supported build chain is quite important.

```suggestion
- Fixed five or more minor bugs.
- A new golang major or minor version has been adopted.
```

Author: > A new golang major or minor version has been adopted.

This should be already covered by `Fixed one or more major CVEs (>=7.5)`. If a new golang major or minor version doesn't have any CVE fixes, then we don't have to necessary release a patch version.

Reviewer: My preference is it can still be worthwhile to cut the patch release even if there is no major CVE as some organisations want to always have software that is built with supported toolchains. When we backporting a golang minor release this does nothing to keep support until the patch release with that new minor version is cut.

Happy to defer to overall consensus and we can leave it for now.

Author: Yes, I agree there is a gray area. The criteria just defines some "hard" requirement. We have some flexibility for golang bumping case. If it has any CVE fixes, then it falls into the category "`Fixed one or more major CVEs (>=7.5)`". Otherwise, we can regard it as a major or minor fix.

But it's open to discussion, and the criteria isn't set of stone. We can always update when needed.
